client/nginx.go

```diff
@@ -378,31 +385,46 @@ type Processes struct {
 	Respawned int64
 }
 
-// NewNginxClient creates an NginxClient.
+// NewNginxClient creates an NginxClient with the latest supported version.
 func NewNginxClient(httpClient *http.Client, apiEndpoint string) (*NginxClient, error) {
+	return NewNginxClientWithVersion(httpClient, apiEndpoint, APIVersion)
+}
+
+//NewNginxClientWithVersion creates an NginxClient with the given version of NGINX Plus API.
+func NewNginxClientWithVersion(httpClient *http.Client, apiEndpoint string, version int) (*NginxClient, error) {
+	if !versionSupported(version) {
+		return nil, fmt.Errorf("API version %v is not supported by the client", version)
+	}
 	versions, err := getAPIVersions(httpClient, apiEndpoint)
 	if err != nil {
 		return nil, fmt.Errorf("error accessing the API: %v", err)
 	}
-
 	found := false
 	for _, v := range *versions {
-		if v == APIVersion {
+		if v == version {
 			found = true
 			break
 		}
 	}
-
 	if !found {
-		return nil, fmt.Errorf("API version %v of the client is not supported by API versions of NGINX Plus: %v", APIVersion, *versions)
+		return nil, ErrUnsupportedVer
```

Comment:
With this change the error message no longer mentions the requested version and the supported versions, is this ok?

Reply:
I think that's ok. 
We have the exported variable with the following docstring: 
```
// APIVersion is the default version of NGINX Plus API supported by the client.
```
`NewNginxClient()`'s docstring is updated accordingly.
```
// NewNginxClient creates an NginxClient with the latest supported version.
```
`NewNginxClientWithVersion()` requires the version as a parameter.
```
//NewNginxClientWithVersion creates an NginxClient with the given version of NGINX Plus API
```
So, the caller has everything to be aware of the version in question.

Reply:
Having `ErrUnsupportedVer` is convenient for implementing a fallback to the previous version, e.g. if the latest or current is not available in running NGINX.